custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java

```diff
@@ -11,44 +11,47 @@
 import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.XML_EXTENSION;
 
 import java.io.File;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
 
 import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
 
 /**
- * Checks whether whitespace characters are use instead of tabs in xml files 
+ * Checks whether whitespace characters are use instead of tabs in xml files
  * indentations and generates warnings in such cases.
- * 
+ *
  * @author Lyubomir Papazov - initial contribution
+ * @author Kristina Simova - Changed regular expression to match lines starting with whitespaces
  *
  */
 public class OnlyTabIdentationInXmlFilesCheck extends AbstractStaticCheck {
 
-    private static final String PATTERN_TO_BE_FOLLOWED = "^\\t*[<].*";
+    private static final Pattern PATTERN_TO_BE_FOLLOWED = CommonUtils.createPattern("[^\\S\\n\\t]+[<]");
```

Comment:
Personally I see no added value for using this util here?

Reply:
Can you also explain what you think that it does

Reply:
So if an empty line contains spaces its not reported
If a line does not have any < but does start with spaces it is not reported and more

What is our goal?

Imho we just should check whether a line starts with spaces? Or maybe contains any spaces before the first non whitespace character.

What about the following:

`^\s* +\s*\S`

Reply:
You are absolutely right, I will fix it right away!